index.bs

```diff
@@ -0,0 +1,550 @@
+
+For example, to route notifications to a user's mobile device when they get up
+to grab a coffee the messaging application could detect that it is no longer
+visible and start sending push messages to the mobile device while marking the
+desktop notifications as "hide on active". If the user were still at their desk
+but using a different application then they would start getting the distracting
+notifications from their mobile device this proposal is attempting to avoid
+whether or not the desktop is able to successfully suppress them. Successful
+suppression of duplicate and disruptive notification requires multi-device
+coordination.
+
+Allowing notifications to be hidden also breaks implementor mitigations for the
+[[PUSH-API]] being used to run silent background tasks.
+
+# Observing User Presence # {#api}
+
+## Model ## {#api-model}
+
+This specification defines a model for user presence on two dimensions: idle
+state and screen lock.
+
+### The {{UserIdleState}} enum ### {#api-useridlestate}
+
+<xmp class="idl">
+enum UserIdleState {
+    "active",
+    "idle"
+};
+</xmp>
+
+: {{"active"}}
+
+::  Indicates that the user has interacted with the device in the last
+    {{IdleOptions/threshold}} milliseconds.
+
+: {{"idle"}}
+
+::  Indicates that the user has no interacted with the device in at least
+    {{IdleOptions/threshold}} milliseconds.
+
+### The {{ScreenIdleState}} enum ### {#api-screenidlestate}
+
+<xmp class="idl">
+enum ScreenIdleState {
+    "locked",
+    "unlocked"
+};
+</xmp>
+
+: {{"locked"}}
+
+::  Indicates that the device has engaged a screensaver or lock screen which
+    prevents content from being seen or interacted with.
+
+: {{"unlocked"}}
+
+::  Indicates that the device is able to display content and be interacted with.
+
+## Permissions ## {#api-permissions}
+
+The <dfn for=PermissionName enum-value>"idle-detection"</dfn> [=powerful
+feature=] is a [=boolean feature=].
+
+## Permissions policy ## {#api-permissions-policy}
+
+This specification defines a [=policy-controlled feature=] identified by the
+string `"idle-detection"`. Its [=default allowlist=] is `["self"]`.
+
+<div class="note">
+The [=default allowlist=] of `["self"]` allows usage of this feature on
+same-origin nested frames by default but prevents access by third-party content.
+
+Third-party usage can be selectively enabled by adding the
+`allow="idle-detection"` attribute to an {{iframe}} element:
+
+<div class="example">
+```html
+<iframe src="https://example.com" allow="idle-detection"></iframe>
+```
+</div>
+
+Alternatively, this feature can be disabled completely in first-party contexts
+by specifying the permissions policy in an HTTP response header:
+
+<div class="example">
+```http
+Permissions-Policy: idle-detection 'none'
+```
+</div>
+
+See [[PERMISSIONS-POLICY]] for more details.
+</div>
+
+## The {{IdleDetector}} interface ## {#api-idledetector}
+
+<xmp class="idl">
+dictionary IdleOptions {
+  [EnforceRange] unsigned long threshold;
```

Comment:
Any reason not to do unsigned long long, per https://w3ctag.github.io/design-principles/#numeric-types ? I'll grant that the advice there isn't followed very much...